csharp/src/SeedLang/Visualization/Value.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021-2022 The SeedV Lab.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using SeedLang.Runtime;
+
+namespace SeedLang.Visualization {
+  // A wrapper class of the underlying VMValue. It's used to represent runtime values in the
+  // notification events of visualizers.
+  public class Value : IEquatable<Value> {
+    public bool IsNil => _value.IsNil;
+
+    public bool IsBoolean => _value.IsBoolean;
+    public bool IsNumber => _value.IsNumber;
+    public bool IsString => _value.IsString;
+
+    public bool IsDict => _value.IsDict;
+    public bool IsList => _value.IsList;
+    public bool IsTuple => _value.IsTuple;
+
+    public int Length => _value.Length;
+
+    private readonly VMValue _value;
+
+    public Value() {
+      _value = new VMValue();
+    }
+
+    public Value(bool boolean) {
+      _value = new VMValue(boolean);
+    }
+
+    public Value(double number) {
+      _value = new VMValue(number);
+    }
+
+    public Value(string str) {
+      _value = new VMValue(str);
+    }
```

Comment:
In case we need a few constructors for the client code to instantiate a value themselves, do we need the constructors to instantiate a value from a list or a dictionary as well? E.g.:

```
    public Value(IReadOnlyList<Value> ...) {
      _value = ...
    }
```

Reply:
The constructors of Tuple, List and Dict might be needed in the future. But I don't know if we really need it now. Let's postpone them?

Reply:
True. We will need constructors for Tuple/List/Dict sometime later but we can postpone it somehow.